test/OrchardCore.Tests/Data/ContentItemTests.cs

```diff
@@ -101,6 +102,27 @@ public void ContentShouldStoreFields()
 
             Assert.Contains(@"""MyPart"":{""Text"":""test"",""myField"":{""Value"":123}}", json);
         }
+
+        [Fact]
+        public void ContentShouldCanCallRemoveMethod()
+        {
+            var contentItem = new ContentItem();
+            contentItem.GetOrCreate<MyPart>();
+            contentItem.Alter<MyPart>(x => x.Text = "test");
+            Assert.Equal("test", contentItem.As<MyPart>().Text);
+            Assert.True(contentItem.Content.Remove("MyPart"));
+        }
+
+        [Fact]
+        public void ContentShouldCanCallSelectNodeMethod()
+        {
+            var contentItem = new ContentItem();
+            contentItem.GetOrCreate<MyPart>();
+            contentItem.Alter<MyPart>(x => x.Text = "test");
+            Assert.Equal("test", contentItem.As<MyPart>().Text);
+            var node = (JsonNode)contentItem.Content.SelectNode("MyPart");
+            Assert.NotNull(node);
+        }
```

Comment:
To clarify the problems in #15505, these should pass if added at the end of this test method:
```csharp
            Assert.NotNull(((JsonObject)contentItem.As<MyPart>().Content).SelectNode("Text")); // fails

            Assert.NotNull((JsonNode)contentItem.Content.SelectNode("MyPart.Text")); // passes
            Assert.NotNull(JObject.Parse(node.ToJsonString()).SelectNode("Text")); // passes
            Assert.NotNull(node.SelectNode("Text")); // fails
```

The first line is what I mentioned at the start of the issue ("If you try to use it on a content part's Content property, it won't work unless you re-serialize the JSON object first").

The other three lines show that chaining "should" work but doesn't. I think the PR only fixes #15505 if these all pass.

Reply:
> you can't use SelectNode to find a path from a child object.

Yes, that's the bug. Like I've showed in the issue, it works with Newtonsoft's SelectToken so it should continue working with STJ as well.